File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map_dict = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement not in map_dict:
                map_dict[nums[i]] = i
            else:
                return [map_dict[complement], i]

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var hash_map = new Map();
    var ans = [];
    for(var i = 0; i < nums.length; i++) {
        var complement = target - nums[i];
        if(hash_map.has(complement)) {
            return [hash_map.get(complement), i];
        }
        hash_map.set(nums[i], i);
    }
    return null;
};